### ISSUE TITLE
Change timeout before sending re-invite

### ISSUE THREAD

**Issue by ben-rowe** [kirm/sip.js#98]
Hi i have an issue where i send an invite and it's taking upwards of 500ms to return which is causing the invite's to be sent again. Is there a way of modifying the re-invite timer in the sip.send() function?

Many Thanks
Ben

**Comment by kirm** [comment]
Do you mean that sip.send blocks for 500ms?

**Comment by ben-rowe** [comment]
yes i do i actually modified it my self on a pull request in the end. Unsure if it is usefull to the project?

**Comment by ben-rowe** [comment]
fork even*

**Comment by kirm** [comment]
sip.send should not be blocking, it should return immediately. Can you provide some more information?

**Comment by ben-rowe** [comment]
Oh no doesnt block i just wanted to modify the timing in the options object. We have a provider giving us LRN and CNAM over SIP. On there dev enviroment they can take over 500ms to complete causing another invite to be sent out. Unfortunatly they dont send out 100 tryings and even though the id sent is the same then then send us back 2 responses. charging us double. I just wanted a way of modifying the timings to prevent this issue i forked and made the modification. didnt know if anyone else had run into this issue or if the mods i made are usefull to anyone else.

**Comment by kirm** [comment]
Well I think i can implement timers configuration in options object you pass to sip.start.
But in my experience fiddling with timers is not a good way to handle a race condition. And given that sip.js uses recommended values of timeouts from sip spec, any client should be experiencing those problems with your provider. So maybe the issue could be resolved some other way? May be your provider need some additional information in INVITEs to map identify them properly?
